src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,95 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
+
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.Todo;
+
+/**
+ * Represents loading data from file and updating data to file
+ */
+public class Storage {
+    private final String filePath;
+
+    /**
+     * Storage constructor
+     * @param filePath
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * method for reading data from saving file
+     * @return tasks list from saving file
+     * @throws IOException
+     */
+    public TaskList load() throws IOException {
+        TaskList tasks = new TaskList();
+        try {
+            File file = new File(filePath);
+            Scanner sc = new Scanner(file);
+            while (sc.hasNext()) {
+                Task task = null;
+                String data = sc.nextLine();
+                String[] commandInFile = data.split(" \\| ");
+                boolean isDoneInFile = commandInFile[1].equals("1");
+                if (commandInFile[0].equals("T")) {
+                    task = new Todo(commandInFile[2], true);
+                    task.isDone = isDoneInFile;
+                    tasks.add(task);
+                } else if (commandInFile[0].equals("D")) {
+                    task = new Deadline(commandInFile[2], commandInFile[3], true);
+                    task.isDone = isDoneInFile;
+                    tasks.add(task);
+                } else if (commandInFile[0].equals("E")) {
+                    task = new Event(commandInFile[2], commandInFile[3], commandInFile[4], true);
+                    task.isDone = isDoneInFile;
+                    tasks.add(task);
+                }
+                if (task != null) {
+                    task.isDone = isDoneInFile;
+                }
+            }
+        } catch (FileNotFoundException e) {
+            new File("data").mkdir();
+            new File(filePath).createNewFile();
+        }
+        return tasks;
+    }
+
+    /**
+     * method for updating date in saving file
+     * @param t
+     * @throws IOException
+     */
+    public void updateFile(TaskList t) throws IOException {
```

Comment:
Single letter argument name (t) might make sense here because the method is short, but maybe we can guard against future confusion by just using "taskList" as the argument name instead since in the future this method might get longer.